Publishing to html5 fails when an image path has query

I have a map with a topic that contains a image element like this: 
`<image href="image.png?width=250" id="image_k31_t2r_qsb"/>`

When I tried to publish the map to html5 using version 3.7.0 and I received the following error: 
`Error: Failed to run pipeline: URI has a query component`

It seems to be related to this issue; #3717

This is a sample DITA map: [sampleDitaOT.zip](https://github.com/dita-ot/dita-ot/files/8108367/sampleDitaOT.zip)

The error occurs here:

    Caused by: java.lang.IllegalArgumentException: URI has a query component
	at java.base/java.io.File.<init>(File.java:429)
	at org.dita.dost.util.URLUtils.exists(URLUtils.java:703)
	at org.dita.dost.module.GenMapAndTopicListModule.categorizeReferenceFile(GenMapAndTopicListModule.java:632)

There are also places in the Store interface like org.dita.dost.store.Store.transform(..) which fail similarly when the URIs have a query and an attempt to create a new file over it throws the error. And I think we should always be flexible in this regard, remove the query, anchor and then attempt to create a file over the URI.

--- a/src/main/java/org/dita/dost/ant/types/JobSourceSet.java
+++ b/src/main/java/org/dita/dost/ant/types/JobSourceSet.java
@@ -25,6 +25,7 @@ import org.dita.dost.util.Constants;
 import org.dita.dost.util.FileUtils;
 import org.dita.dost.util.Job;
 import org.dita.dost.util.Job.FileInfo;
+import org.dita.dost.util.URLUtils;
 
 import java.io.File;
 import java.net.MalformedURLException;
@@ -63,7 +64,7 @@ public class JobSourceSet extends AbstractFileSet implements ResourceCollection 
                     log("Found temporary directory file " + tempFile, Project.MSG_VERBOSE);
                     res.add(new StoreResource(job, job.tempDirURI.relativize(f.uri)));
                 } else if (f.src != null && Objects.equals(f.src.getScheme(), "file")) {
-                    final File srcFile = new File(f.src);
+                    final File srcFile = new File(URLUtils.setQuery(f.src, null));
                     if (srcFile.exists()) {
                         log("Found source directory file " + srcFile, Project.MSG_VERBOSE);
                         final File rel = FileUtils.getRelativePath(new File(new File(job.getInputDir()), "dummy"), srcFile);

--- a/src/main/java/org/dita/dost/util/URLUtils.java
+++ b/src/main/java/org/dita/dost/util/URLUtils.java
@@ -564,6 +564,21 @@ public final class URLUtils {
     }
 
     /**
+     * Create new URI with a given query.
+     *
+     * @param path URI to set query on
+     * @param query new query, {@code null} for no query
+     * @return new URI instance with given query
+     */
+    public static URI setQuery(final URI path, final String query) {
+        try {
+            return new URI(path.getScheme(), path.getUserInfo(), path.getHost(), path.getPort(), path.getPath(), query, path.getFragment());
+        } catch (final URISyntaxException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    /**
      * Remove fragment from URI.
      *
      * @param path remove fragment from this URI
@@ -705,7 +720,8 @@ public final class URLUtils {
         if (file.getScheme() == null) {
             return new File(file.getPath()).exists();
         } else if ("file".equals(file.getScheme())) {
-            return new File(file).exists();
+            final URI resource = removeFragment(setQuery(file, null));
+            return new File(resource).exists();
         } else {
             // Assume non-file URIs always exists and force fetching them
             return true;

--- a/src/main/java/org/dita/dost/writer/ValidationFilter.java
+++ b/src/main/java/org/dita/dost/writer/ValidationFilter.java
@@ -197,7 +197,7 @@ public final class ValidationFilter extends AbstractXMLFilter {
         if (href != null) {
             try {
                 final URI uri = new URI(href);
-                final URI abs = URLUtils.stripFragment(currentFile.resolve(uri)).normalize();
+                final URI abs = URLUtils.setQuery(URLUtils.stripFragment(currentFile.resolve(uri)).normalize(), null);
                 if (abs.getScheme() != null && abs.getScheme().equals("file")) {
                     final File p = new File(abs);
                     try {

--- a/src/test/java/org/dita/dost/module/TestGenMapAndTopicListModule.java
+++ b/src/test/java/org/dita/dost/module/TestGenMapAndTopicListModule.java
@@ -26,6 +26,7 @@ import static org.dita.dost.util.Constants.*;
 import static org.dita.dost.util.Job.FileInfo;
 import static org.dita.dost.util.Job.Generate.NOT_GENERATEOUTTER;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
 
 public class TestGenMapAndTopicListModule {
 
@@ -326,6 +327,18 @@ public class TestGenMapAndTopicListModule {
                         .filter(f -> !f.isResourceOnly)
                         .map(fi -> fi.uri.toString())
                         .collect(Collectors.toSet()));
+    }
+
+    @Test
+    public void testImage() throws Exception{
+        final File inputDir = new File("image");
+        final File inputMap = new File(inputDir, "image.dita");
+        final File outDir = new File(tempDir, "out");
+
+        final Job job = generate(inputDir, inputMap, outDir, tempDir);
+
+        assertNotNull(job.getFileInfo(URI.create("image.svg")));
+        assertNotNull(job.getFileInfo(URI.create("image.svg?media=print")));
     }
 
     private Set<String> readLines(final File f) throws IOException {

--- a/src/test/java/org/dita/dost/util/URLUtilsTest.java
+++ b/src/test/java/org/dita/dost/util/URLUtilsTest.java
@@ -283,4 +283,12 @@ public class URLUtilsTest {
         final URI act = toDirURI(src);
         assertTrue(act.toString().endsWith("/"));
     }
+
+    @Test
+    public void setQuery() {
+        assertEquals(URI.create("foo"), URLUtils.setQuery(URI.create("foo"), null));
+        assertEquals(URI.create("foo?baz"), URLUtils.setQuery(URI.create("foo"), "baz"));
+        assertEquals(URI.create("foo"), URLUtils.setQuery(URI.create("foo?bar"), null));
+        assertEquals(URI.create("foo?baz"), URLUtils.setQuery(URI.create("foo?bar"), "baz"));
+    }
 }

--- a/src/test/resources/TestGenMapAndTopicListModule/src/image/image.dita
+++ b/src/test/resources/TestGenMapAndTopicListModule/src/image/image.dita
@@ -1,0 +1,9 @@
+<topic xmlns:ditaarch="http://dita.oasis-open.org/architecture/2005/" class="- topic/topic "
+       domains="(topic abbrev-d)                            a(props deliveryTarget)                            (topic equation-d)                            (topic hazard-d)                            (topic hi-d)                            (topic indexing-d)                            (topic markup-d)                            (topic mathml-d)                            (topic pr-d)                            (topic relmgmt-d)                            (topic sw-d)                            (topic svg-d)                            (topic ui-d)                            (topic ut-d)                            (topic markup-d xml-d)   "
+       id="topic_fnd_3l4_xz" ditaarch:DITAArchVersion="1.3">
+  <title class="- topic/title "></title>
+  <body class="- topic/body ">
+    <image class="- topic/image " href="image.svg"/>
+    <image class="- topic/image " href="image.svg?media=print#foo"/>
+  </body>
+</topic>

